cmd/metal-api/main.go

```diff
@@ -710,7 +712,7 @@ func initRestServices(withauth bool) *restfulspec.Config {
 	restful.DefaultContainer.Filter(metrics.RestfulMetrics)
 
 	if withauth {
-		restful.DefaultContainer.Filter(rest.UserAuth(initAuth(lg.Sugar())))
+		restful.DefaultContainer.Filter(rest.UserAuth(userGetter))
```

Comment:
What happens if main is called withoutAuth ?

Reply:
I think this is really just a branch for making the spec generation work. As soon as you get the spec generation and this feature working, everything should be at the right place.

Reply:
make spec does not work, metal-api.json is empty, this should also be checked in ci that no changes to metal-api.json are created and fails if so.

Reply:
added a CI job which ensures spec was generated and commited.

Reply:
does not work

Reply:
works now